codegenerator/cli/templates/static/codegen/src/globalState/GlobalState.res

```diff
@@ -913,8 +913,11 @@ let injectedTaskReducer = (
     }
   | PreRegisterDynamicContracts =>
     if !state.currentlyProcessingBatch && !isRollingBack(state) {
-      switch state.chainManager->ChainManager.createBatch(~maxBatchSize=state.maxBatchSize) {
-      | {isInReorgThreshold: false, val: Some({batch, fetchStatesMap, arbitraryEventQueue})} =>
+      switch state.chainManager->ChainManager.createBatch(
+        ~maxBatchSize=state.maxBatchSize,
+        ~onlyBelowReorgThreshold=true,
+      ) {
+      | {val: Some({batch, fetchStatesMap, arbitraryEventQueue})} =>
```

Comment:
For preregistration, we only want to create batches below the reorg threshold.

Then as long as "Some" batch is returned we process it.